File: puzzles/cs/ascii-art/ascii-art.cs
using System;

class Program
{
    static void Main(string[] args)
    {
        int width = int.Parse(Console.ReadLine());
        int height = int.Parse(Console.ReadLine());
        string text = Console.ReadLine().ToUpper();

        for (int i = 0; i < height; i++)
        {
            string row = Console.ReadLine();
            string output = "";

            foreach (char character in text)
            {
                int position = character - 'A';

                if (position < 0 || position > 25)
                    position = 26;

                int start = position * width;
                int end = start + width;
                output += row.Substring(start, width);
            }

            Console.WriteLine(output);
        }
    }
}

File: puzzles/ts/ascii-art/ascii-art.ts
const width: number = Number(readline());
const height: number = Number(readline());
const text: string = readline().toUpperCase();

for (let i: number = 0; i < height; i++) {
  const row: string = readline();
  let output: string = "";
  for (let letter of text) {
    let alphabetPosition = letter.charCodeAt(0) - 'A'.charCodeAt(0);
    if (alphabetPosition < 0 || alphabetPosition > 25) {
      alphabetPosition = 26;
    }
    let column = alphabetPosition * width;
    output += row.substring(column, column + width);
  }
  console.log(output);
}
